### PR TITLE
Fix colorbar templating for "non-root" colorscales

### PR DESCRIPTION
**Pull request by etpinard**
fixes https://github.com/plotly/plotly.js/issues/4467

before: https://codepen.io/etpinard/pen/ZEYrLdo
after: https://codepen.io/etpinard/pen/zYxRNVm (`colorbar.ypad` now works)

cc @archmoj @wbrgss

### CODE REVIEW

File: test/image/compare_pixels_test.js [hunk 1]
```diff
@@ -101,6 +101,7 @@ if(allMock || argv.filter) {
 }
 
 var FLAKY_LIST = [
+    'treemap_coffee',
```
Comment (etpinard) [inline test/image/compare_pixels_test.js]: cc @archmoj something is off with this new mock. See https://circleci.com/gh/plotly/plotly.js/64411?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link - it failed 5 times in a row.

Comment (archmoj) [reply test/image/compare_pixels_test.js]: You can pick https://github.com/plotly/plotly.js/commit/0175108c7670910ad2175367767381397ccd3779 to fix that.

**Comment by archmoj** [comment]
Thank you!
:dancer:

**Comment by wbrgss** [comment]
Thanks for the quick turnaround @etpinard!
The other `colorbar` attributes (e.g. `title`) appear to be working in templates now also :tada:

**Pull request merged**